packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart

```diff
@@ -14,9 +16,174 @@ part 'sk_payment_queue_wrapper.g.dart';
 /// The payment queue contains payment related operations. It communicates with App Store and presents
 /// a user interface for the user to process and authorize the payment.
 class SKPaymentQueueWrapper {
+  SKTransactionObserverWrapper _observer;
+
+  /// Returns the default payment queue.
+  ///
+  /// We do not support instantiating a custom payment queue, hence the singleton. However, you can
+  /// override the observer
+  factory SKPaymentQueueWrapper() {
+    return _singleton;
+  }
+
+  static final SKPaymentQueueWrapper _singleton = new SKPaymentQueueWrapper._();
+
+  SKPaymentQueueWrapper._() {
+    callbackChannel.setMethodCallHandler(_handleObserverCallbacks);
+  }
+
   /// Calls [`-[SKPaymentQueue canMakePayments:]`](https://developer.apple.com/documentation/storekit/skpaymentqueue/1506139-canmakepayments?language=objc).
   static Future<bool> canMakePayments() async =>
       await channel.invokeMethod('-[SKPaymentQueue canMakePayments:]');
+
+  /// Sets a transaction observer to listen to all the transaction events of the payment queue.
+  ///
+  /// You have to have a transaction observer to be added to the payment queue to handle the
+  /// payment follow.
+  /// You must set the observer right when App launches to avoid missing callback when your user
+  /// started a purchase flow from the App Store.
+  void setTransactionObserver(SKTransactionObserverWrapper observer) {
```

Comment:
Is there any way we could take this as a param at construction time instead? It would force users to set this as soon as this object is instantiated.

Reply:
Since the payment queue is a singleton, there is no actually a construction time. User can technically call SKPaymentQueue() everywhere to refer the defaultPaymentQueue. So I didn't want user to have to put the observer each time when they refer to the singleton.

Reply:
Right. Hmm. I think it may be worth converting this from a singleton into a constructed class just to enforce that this is configured ASAP. What do you think? Could be done in a followup PR too.

Reply:
Let's do more offline discussion on this and will create another PR if we reach a different approach.